modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/handler/MediaHandler.java

```diff
@@ -175,19 +175,26 @@ private void verifyMediaScope( final String projectContext, final RepositoryId r
         {
             if ( MEDIA_SCOPE_DELIMITER_PATTERN.splitAsStream( mediaServiceScope ).map( String::trim ).noneMatch( projectContext::equals ) )
             {
-                throw WebException.notFound( "Not a valid media url pattern" );
+                throw createNotFoundException();
             }
         }
-        else
+
+        if ( webRequest.getRawPath().startsWith( "/api/" ) )
         {
-            if ( webRequest instanceof final PortalRequest portalRequest )
-            {
-                if ( portalRequest.isSiteBase() &&
-                    !( repositoryId.equals( portalRequest.getRepositoryId() ) && branch.equals( portalRequest.getBranch() ) ) )
-                {
-                    throw WebException.notFound( "Not a valid media url pattern" );
-                }
-            }
+            return;
+        }
+
+        if ( !( webRequest.getRawPath().startsWith( "/site/" ) || webRequest.getRawPath().startsWith( "/admin/site/" ) ) )
+        {
+            throw createNotFoundException();
+        }
+
```

Comment:
By this time web request must be already portal request. Creating new instance to then throw it away is a waste